teacher.rb

```diff
@@ -0,0 +1,16 @@
+require './person'
+
+class Teacher < Person
+  attr_reader :specialization, :name, :age
+
+  def initialize(name, age, specialization)
+    super(age, name)
+    @specialization = specialization
+    @name = name
+    @age = age
+  end
```

Comment:
- [ ] Kindly add `parent_permission = true` in your constructor because it is inherited from the parent class Person and it is required to pass it.
- [ ] As stated above, it is not needed to add 
```ruby
    @name = name
    @age = age
```